python/cog/server/response_throttler.py

```diff
@@ -0,0 +1,22 @@
+import time
+import os
+
+from ..response import Status
+
+
+class ResponseThrottler:
+    def __init__(self):
+        self.last_sent_response_time = 0
+        self.response_interval = float(
+            os.environ.get("COG_THROTTLE_RESPONSE_INTERVAL", 0)
+        )
+
+    def should_send_response(self, response):
+        if Status.is_terminal(response["status"]):
+            return True
+
+        seconds_since_last_response = time.time() - self.last_sent_response_time
```

Comment:
Nit: my instinct is to make `seconds_since_last_response` a method (possibly private).